static/app/views/performance/transactionSummary/transactionSpans/spanDetails/spanDetailsSearchBar.tsx

```diff
@@ -0,0 +1,50 @@
+import {browserHistory} from 'react-router';
+import styled from '@emotion/styled';
+import {Location} from 'history';
+
+import SearchBar from 'sentry/components/events/searchBar';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {Organization} from 'sentry/types';
+import EventView from 'sentry/utils/discover/eventView';
+import {decodeScalar} from 'sentry/utils/queryString';
+
+interface SpanDetailsSearchBarProps {
+  eventView: EventView;
+  location: Location;
+  organization: Organization;
+}
+
+export default function SpanDetailsSearchBar({
+  organization,
+  eventView,
+  location,
+}: SpanDetailsSearchBarProps) {
+  const query = decodeScalar(location.query.query, '');
+
+  const handleSearchQuery = (searchQuery: string): void => {
+    browserHistory.push({
+      pathname: location.pathname,
+      query: {
+        ...location.query,
+        cursor: undefined,
+        query: String(searchQuery).trim() || undefined,
+      },
+    });
+  };
+
+  return (
+    <StyledSearchBar
+      placeholder={t('Filter Transactions')}
+      organization={organization}
+      projectIds={eventView.project}
+      query={query}
+      fields={eventView.fields}
+      onSearch={handleSearchQuery}
+    />
+  );
+}
+
+const StyledSearchBar = styled(SearchBar)`
+  margin-bottom: ${space(2)};
+`;
```

Comment:
If you really wanna get rid of `StyledSearchBar` you could write it as

```suggestion
const SpanDetailsSearchBar = styled(({
  organization,
  eventView,
  location,
  className,
}: SpanDetailsSearchBarProps) => {
  const query = decodeScalar(location.query.query, '');

  const handleSearchQuery = (searchQuery: string): void => {
    browserHistory.push({
      pathname: location.pathname,
      query: {
        ...location.query,
        cursor: undefined,
        query: String(searchQuery).trim() || undefined,
      },
    });
  };

  return (
    <SearchBar
      className={className}
      placeholder={t('Filter Transactions')}
      organization={organization}
      projectIds={eventView.project}
      query={query}
      fields={eventView.fields}
      onSearch={handleSearchQuery}
    />
  );
})`
  margin-bottom: ${space(2)};
`;

export default SpanDetailsSearchBar
```

but I'm indifferent

Reply:
hm I'd prefer here to create a separate styled component to separate concerns a bit so component styles and logic are visibly separated

Reply:
One functional difference here to recognize btw, is when you do what I suggested, you can continue to use it with `styled(SpanDetailsSearchBar)` or as a selector in other styled components. (Not that you would need to do that here, but worth being aware of!)

Reply:
Good to know 🤔